spring-integration-jms/src/main/java/org/springframework/integration/jms/JmsOutboundGateway.java

```diff
@@ -659,6 +700,19 @@ protected Object handleRequestMessage(final Message<?> message) {
 				jmsReply = this.sendAndReceiveWithoutContainer(requestMessage);
 			}
 			else {
+				if (this.idleReplyContainerTimeout > 0) {
+					synchronized(this.lifeCycleMonitor) {
+						this.lastSend = System.currentTimeMillis();
+						if (!this.replyContainer.isRunning()) {
+							if (logger.isDebugEnabled()) {
+								logger.debug(this.getComponentName() + ": Starting reply container.");
+							}
+							this.replyContainer.start();
+							this.idleTask = getTaskScheduler().scheduleAtFixedRate(new IdleContainerStopper(),
+									this.idleReplyContainerTimeout / 2);
```

Comment:
Gary, would you mind explaining a bit more about this solution?
Why isn't `scheduleWithFixedDelay` and why `this.idleReplyContainerTimeout / 2` ?